### DISCUSSION TITLE
Transferring Github to D drive

### DISCUSSION THREAD

**Discussion by BSSE23011** [community/community#66767] [Programming Help] [answered]
Hey Everyone!!!!! I downloaded Github on Windows 10 and it by default installed on c drive. My c drive is getting full  so i want to transfer github to d drive. i discovered that github had created three folders on c drive. One folder is in the "Local" Folder, Second is in the "Roaming" Folder and the last folder containing the repositories in Documents. so i want these all things to be securely transferred to D drive.

I want the same thing for C Lion....
Just want that both of these softwares dont take a single bit of C drive
Thank you very much!!!

**Comment by theintrigued** [comment, marked as answer]
Here are the steps, if I understand correctly:

- **Backup Your GitHub Repositories**:
Before making any changes, it's crucial to back up your GitHub repositories to ensure you don't lose any data. You can do this by copying the repositories from the current location to a safe place on your D drive or an external storage device.

- **Uninstall GitHub Desktop (optional but recommended)**:
If you have GitHub Desktop installed, you can uninstall it. This will not delete your repositories but will remove the GitHub Desktop application.

- **Move GitHub Configuration Files**:
GitHub configuration files are usually located in your user directory. Follow these steps to move them to your D drive:
Open File Explorer and go to C:\Users\<YourUsername>\.gitconfig. 
Copy the .gitconfig file. 
Navigate to your D drive (e.g., D:\) and paste the .gitconfig file here.

- **Update Environment Variables (if necessary)**:
Some environment variables may point to the C drive. To check and update them:
Search for "Environment Variables" in the Windows Start menu and click on "Edit the system environment variables."
In the System Properties window, click the "Environment Variables" button.
Under "User variables" and "System variables," look for any paths or variables that point to the old GitHub location (typically on the C drive). If you find any, update them to point to the new location on your D drive.

- **Move GitHub Repositories**:
Now, you can move the GitHub repositories from your C drive to your D drive. Typically, your repositories are stored in the Documents folder. Here's how to do it:
Open File Explorer.
Navigate to C:\Users\<YourUsername>\Documents\GitHub.
Copy the entire GitHub folder. Go to your D drive (e.g., D:\) and paste the GitHub folder there.

- **Update GitHub Desktop (if you re-install)**: 
If you choose to re-install GitHub Desktop on your D drive, make sure to set the default repository location to the new location on your D drive during installation.

- **Reconfigure GitHub Desktop (if you re-install**):
If you reinstall GitHub Desktop, you may need to reconfigure your accounts and repository settings within the application.

- **Restart Your Computer**:
To ensure that all changes take effect, restart your computer.

Hope this helps!

**Reply by BSSE23011** [reply]
So, following this, github is not gonna take a single bit from c drive, right?
can you please tell me how to do this all for c lion?

**Reply by theintrigued** [reply]
Well , I cannot guarantee a success due to the fact that I am not aware of your system settings/the way things are set up for you, but I'd suggest its worth a try (**after taking a backup of the files**).

And regarding C Lion, I have not worked with C Lion at all, so i'm in the dark on that topic.

**Reply by BSSE23011** [reply]
Well, Thank you very much Sir!!

**Reply by theintrigued** [reply]
Please do mark the post as the answer if this helped you!
So others can find it easily too ❤️

**Comment by pomjoep** [comment]
When i reinstall it there is no options to choose where it installs

**Comment by emilianowilson58** [comment]
I like more
